packages/store/src/decorators/state.ts

```diff
@@ -1,42 +1,28 @@
-import { ensureStoreMetadata } from '../internal/internals';
-import { StoreOptions, META_KEY } from '../symbols';
-
-const stateNameRegex = new RegExp('^[a-zA-Z0-9_]+$');
-
-/**
- * Error message
- * @ignore
- */
-export const stateNameErrorMessage = (name: string) =>
-  `${name} is not a valid state name. It needs to be a valid object property name.`;
+import { ensureStoreMetadata, MetaDataModel, StateClass } from '../internal/internals';
+import { META_KEY, META_OPTIONS_KEY, StoreOptions } from '../symbols';
+import { StoreValidators } from '../utils/store-validators';
 
 /**
  * Decorates a class with ngxs state information.
  */
 export function State<T>(options: StoreOptions<T>) {
-  return function(target: any) {
-    const meta = ensureStoreMetadata(target);
+  return function(target: StateClass) {
+    const meta: MetaDataModel = ensureStoreMetadata(target);
+    const targetReference: StateClass = Object.getPrototypeOf(target);
+    const inheritanceOptions: Partial<StoreOptions<T>> = target[META_OPTIONS_KEY] || {};
+    const { children, defaults, name } = { ...inheritanceOptions, ...options };
```

Comment:
Is it valid to inherit the children? This might lead to a child state share between two top level states which has some caveats. See discussion in issue #494

Reply:
we can cover with tests such a case, just give an example

Reply:
all properties should be inherited, since the essence of the task in this

Reply:
The `ChildSate` will then 'exist' in multiple places in the tree but will have only one MappedStore representation. This will result in the action handlers only being executed for the first place that it was added and the `store.select(ChildState)` only being able to select from only that one state.
I guess that this is an existing issue so it does not need to be addressed by this PR, but this PR makes it more likely that this issue will be encountered so it pushes the priority up for solving it. This would be encountered when you inherit multiple times from a state that has a child state. I guess that the duplicate check may catch this scenario, but we would have to test it. Could you write a test for this?